platform/default/mbgl/map/map_snapshotter.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <mbgl/util/image.hpp>
+#include <mbgl/util/thread.hpp>
+#include <mbgl/util/optional.hpp>
+
+#include <exception>
+#include <string>
+#include <functional>
+
+namespace mbgl {
+
+template<class> class ActorRef;
+struct CameraOptions;
+class FileSource;
+class Size;
+class LatLngBounds;
+
+class SnapshotCallback {
```

Comment:
I left the wrapper class as it makes clear on which thread the callback is going to happen when requiring an `ActorRef` in `MapSnapshotter::snapshot`. If we require only the lambda, we would still internally use an actor/mailbox to ensure the callback happens on the original thread right?